src/main/java/org/tron/core/db/Manager.java

```diff
@@ -1588,8 +1632,18 @@ public boolean isGeneratingBlock() {
   private static class ValidateSignTask implements Callable<Boolean> {
 
     private TransactionCapsule trx;
+    private List<TransactionCapsule> trxList;
     private CountDownLatch countDownLatch;
     private Manager manager;
+    private ByteArrayWrapper owner;
+
+    ValidateSignTask(List<TransactionCapsule> trxList, CountDownLatch countDownLatch,
+        Manager manager, ByteArrayWrapper owner) {
+      this.trxList = trxList;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "trxList". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1916&issues=AWgnKDa4rt4Kct_4Nf2D&open=AWgnKDa4rt4Kct_4Nf2D)

[ISSUE_KEY]: AWgnKDa4rt4Kct_4Nf2D